src/DIRAC/Core/Utilities/CGroups2.py

```diff
@@ -0,0 +1,326 @@
+        OOM, which may allow a process to use more than the memory limit).
+        This will raise a RuntimeError if setting up the slot fails.
+        """
+        if not self._ready:
+            return
+        self._create_group(slot_name)
+        self._set_limits(slot_name, cores, memory, noswap)
+
+    def _remove_slot(self, slot_name):
+        """Removes a slot with the given name.
+        Can raise usual filesystem OSError if the slot doesn't exist.
+        """
+        if not self._ready:
+            return
+        self._remove_group(slot_name)
+
+    def _setup_subproc(self, slot_name):
+        """A subprocess preexec function for setting up cgroups.
+        This will move te current process into the given cgroup slot.
+        On failure, no error will be reported.
+        """
+        # Threading danger!
+        # There are potential threading issues with preexec functions
+        # They must not hold any locks that the parent process might already
+        # be holding, including ones in standard library functions.
+        # This function should be kept as minimal as possible.
+        try:
+            self._write_control(slot_name, self.CTRL_PROCS, f"{os.getpid()}")
+        except Exception as err:
+            # We can't even really log here as we're in the set-up
+            # context of the new proces
+            pass
+
+    def setUp(self):
+        """Creates the base cgroup tree if possible. Should be called once
+        per process before using systemCall.
+        Returns S_OK/S_ERROR.
+        """
+        try:
+            self._prepare()
+        except Exception as err:
+            # The majority of CGroup failures will be RuntimeError
+            # However we don't want any unexpected failure to crash the upstream module,
+            # We just want to continue without cgroup support instead
+            return S_ERROR(str(err))
+        return S_OK()
+
+    def systemCall(self, *args, **kwargs):
+        """A proxy function for Subprocess.systemCall but will create a cgroup2 slot
+        if the functionality is available. An optional ceParameters dictionary
+        may be included, which will be searched for specific cgroup memory options.
+        Returns the usual S_OK/S_ERROR from Subprocess.systemCall.
+        """
+        preexec_fn = None
+        slot_name = f"subproc_{os.getpid()}_{self._subproc_num}"
+        self._subproc_num += 1
+        if self._ready:
+            self.log.info(f"Creating slot cgroup {slot_name}")
+            memory = None
+            noswap = False
+            if "ceParameters" in kwargs:
+                memoryMB = int(kwargs["ceParameters"].get("MemoryLimitMB", 0))
+                if memoryMB:
+                    memory = memoryMB * 1024 * 1024
+                if kwargs["ceParameters"].get("MemoryNoSwap", "no").lower() in ("yes", "true"):
+                    noswap = True
```

Comment:
Can you document these new options in `docs/source/AdministratorGuide/Resources/computingelements.rst`?